test/local/db.js

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const LIB_DIR = '../../lib'
+
+const assert = require('insist')
+const mocks = require('../mocks')
+const P = require(`${LIB_DIR}/promise`)
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+
+const tokenLifetimes = {
+  sessionTokenWithoutDevice: 2419200000
+}
+
+describe('db:', () => {
+  let results, pool, redis, log, tokens, db
+
+  beforeEach(() => {
+    results = {}
+    pool = {
+      get: sinon.spy(() => P.resolve(results.pool)),
+      post: sinon.spy(() => P.resolve()),
+      put: sinon.spy(() => P.resolve())
+    }
+    redis = {
+      getAsync: sinon.spy(() => P.resolve(results.redis)),
+      setAsync: sinon.spy(() => P.resolve())
+    }
+    log = mocks.spyLog()
+    tokens = require(`${LIB_DIR}/tokens`)(log, { tokenLifetimes })
+    const DB = proxyquire(`${LIB_DIR}/db`, {
+      './pool': function () { return pool },
+      redis: { createClient: () => redis }
+    })({ tokenLifetimes, redis: {} }, log, tokens, {})
+    return DB.connect({})
+      .then(result => db = result)
+  })
+
+  describe('sessions:', () => {
```

Comment:
I couldn't come up with a way to test the new behaviour in `db.sessions` remotely, because of the need to either set `createdAt` in the past or to mock the result from `Date.now()`. Having learned my lesson from the multi-year debacle that arose when we used the `_createdAt` query param to do something similar before, this time I've opted for a local test instead.